src/controllers/UserController.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response } from "express";
+import { userRepository } from "../repositories/userRepository";
+
+export class UserController {
+  async signup(req: Request, res: Response) {
+    const { name, email, password } = req.body;
+
+    if (!name || !email || !password) {

```

Comment:
você pode usar o conceito de schemas para verificar a estrutura do seu request. basicamente vc define como vai ser a estrutura do seu request, quais campos vão ter, os tipos de cada campo, quais são required... e coloca isso como um middleware nessa rota. então todo request pra essa rota ele vai verificar se os campos correspondem ao seu schema. 
Dá uma olhada pra implementar isso.

Reply:
Resolvido no commit Validation for user schema